lib/omnibus/util.rb

```diff
@@ -65,18 +99,54 @@ def shellout(*args)
         options[:environment] = options.fetch(:environment, {}).merge(options[:env])
       end
 
+      # Double check that we don't have conflicting PATH definitions.
+      path_keys = options[:environment].keys.grep(/\Apath\Z/i)
+      if path_keys.length > 1
+        raise InvalidValue.new("shellout", "receive environment without duplicate PATH values")
+      elsif path_keys.length == 1 && path_keys.first != path_key
+        options[:environment][path_key] = options[:environment].delete(path_keys.first)
+      end
+
+      # Try our best to get rid of the "outer" omnibus ruby and any current
+      # chef-client/chef-dk installations from the path.
+      if options.delete(:clean_ruby_path)
```

Comment:
Why is this needed? We already do clean the outer Ruby environment:
https://github.com/chef/omnibus/blob/29d390ec93709ceec2667038aed6769ee29a0646/lib/omnibus/builder.rb#L833
https://github.com/chef/omnibus/blob/29d390ec93709ceec2667038aed6769ee29a0646/lib/omnibus/builder.rb#L877-L911